### DISCUSSION TITLE
"@types/node": "15.12.2". Class return from function. TLS.Socket and TLS.Server. No event "error" and others events in types.

### DISCUSSION THREAD

**Discussion by yangu-hury** [DefinitelyTyped/DefinitelyTyped#53726] [Issues with a @types package] [answered]
Package @types/node. 

Versions:
"@types/node": "15.12.2",
"typescript": "4.3.2"

Classes return from function createServer, connect and it callback. TLS.Socket and TLS.Server. No event "error" and others events in types for the "tls" module. It works like expected with "net" module when a event "error" is passed to "tls" in one of the examples in image. But it don't exist in types and in node.js documentation.

note: Node.js documentation is the origin of the problem and types are exactly like node.js docs. It is the reason of the problem. 

![CTPcmO1izI](https://user-images.githubusercontent.com/36553429/121107423-4806bb80-c7de-11eb-8016-fa5b046da6b1.png)
![g2HC89nB0u](https://user-images.githubusercontent.com/36553429/121107425-49d07f00-c7de-11eb-90bc-37f29e72e3c7.png)
![U2cXiwQKxe](https://user-images.githubusercontent.com/36553429/121107426-4a691580-c7de-11eb-8581-d31f02d97c40.png)
![chrome_SrQ44bT6PY](https://user-images.githubusercontent.com/36553429/121107429-4b01ac00-c7de-11eb-9453-b4c4d5e4a693.png)

![tjh1RQuQ3b](https://user-images.githubusercontent.com/36553429/121107554-813f2b80-c7de-11eb-9c6e-30487d197457.png)
![k6oZCCYGKk](https://user-images.githubusercontent.com/36553429/121107563-8308ef00-c7de-11eb-9015-4606c9548e92.png)

![ZCyTbeGwdP](https://user-images.githubusercontent.com/36553429/121107562-82705880-c7de-11eb-9d75-8cce7732504a.png)

**Comment by SimonSchick** [comment, marked as answer]
This is an inherent problem with the EventListener inheritance.

It is very difficult to model all the possible overloads as defining them overwrites inherited overloads which means we'd have to copy/paste all of them for every inheritance scenario.
